control/plugin/client/native.go

```diff
@@ -92,8 +94,47 @@ func (p *PluginNativeClient) Kill(reason string) error {
 	return err
 }
 
-func (p *PluginNativeClient) Publish(contentType string, content []byte, config map[string]ctypes.ConfigValue) error {
-	args := plugin.PublishArgs{ContentType: contentType, Content: content, Config: config}
+func encodeMetrics(metrics []core.Metric) []byte {
+	mts := make([]plugin.MetricType, len(metrics))
+	for i, m := range metrics {
+		mts[i] = plugin.MetricType{
+			Namespace_:          m.Namespace(),
+			Tags_:               m.Tags(),
+			Timestamp_:          m.Timestamp(),
+			Version_:            m.Version(),
+			Config_:             m.Config(),
+			LastAdvertisedTime_: m.LastAdvertisedTime(),
```

Comment:
I added [checkTime](https://github.com/intelsdi-x/snap/pull/1150/files#diff-71933e42357a4d1aa25a10d72c75a57dR100) which is used to do a similar check for both encode/decode metrics. Does this cover that issue?